recipe-server/client/control_new/components/recipes/DetailsActionBar.js

```diff
@@ -0,0 +1,63 @@
+import { Button } from 'antd';
+import { Map } from 'immutable';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'redux-little-router';
+
+import { getRecipe } from 'control_new/state/recipes/selectors';
+import { isLatestApprovedRevision } from 'control_new/state/revisions/selectors';
+import { getUrlParam, getUrlParamAsInt } from 'control_new/state/router/selectors';
+
+
+@connect(
+  state => {
+    const recipeId = getUrlParamAsInt(state, 'recipeId');
+    const revisionId = getUrlParam(state, 'revisionId');
+
+    return {
+      isLatestApproved: isLatestApprovedRevision(state, revisionId),
+      recipe: getRecipe(state, recipeId, new Map()),
+      recipeId,
+      revisionId,
+    };
+  },
+)
+export default class DetailsActionBar extends React.Component {
+  static propTypes = {
+    isLatestApproved: PropTypes.bool,
+    recipe: PropTypes.object,
+    recipeId: PropTypes.number,
+    revisionId: PropTypes.string,
```

Comment:
The new lint rules in #857 is going to complain that these aren't required and also don't have defaults. I suggest making them required, since the selector always passed them in.